src/DefaultBrowserSource.ts

```diff
@@ -0,0 +1,35 @@
+import Evented from 'dojo-core/Evented';
+import global from 'dojo-core/global';
+import { NavigationArgs, RouterSource } from './routing';
+
+/**
+ * The `DefaultBrowserSource` navigates via full page refreshes (the default method of
+ * navigation).
+ *
+ * This allows `Router` to be used where the HTML History API is not supported, and
+ * preserving the URL is preferred over something like a `hashchange` implementation.
+ */
+export default class DefaultBrowserSource extends Evented implements RouterSource {
+	protected _current: string;
+
+	get currentPath(): string {
+		return this._current || global.location.pathname;
+	}
+
+	destroy(): void {
+		this.go = function () {};
+	}
+
+	go(path: string = null): void {
+		if (path === null || !this._current) {
```

Comment:
@kitsonk You're right; I'm going to have to think through a better implementation. The idea here is that if `Router` calls `RouterSource#go` without a path, then the `RouterSource` will get the default path from the enviroment and emit an event that will trigger `Router#go`. In the manual tests I've done this works fine, but as you pointed out this is brittle. I have a working implementation that's more solid, and I'll update this PR once the tests are in place.